app/hotels/src/filter/price/PriceFilter.js

```diff
@@ -188,28 +188,12 @@ const calculateDaysOfStay = (checkin, checkout) => {
   return null;
 };
 
-export default class PriceFilterWithContext extends React.Component<Props> {
-  renderInner = ({
-    currency,
-    checkin,
-    checkout,
-    currentSearchStats,
-  }: HotelsContextState) => {
-    const daysOfStay = calculateDaysOfStay(checkin, checkout);
-    if (daysOfStay === null) {
-      return null;
-    }
-    return (
-      <PriceFilter
-        {...this.props}
-        currentSearchStats={currentSearchStats}
-        currency={currency}
-        daysOfStay={daysOfStay}
-      />
-    );
-  };
+const select = ({ currency, checkin, checkout, currentSearchStats }) => ({
+  currency,
+  checkin,
+  checkout,
+  currentSearchStats,
+  daysOfStay: calculateDaysOfStay(checkin, checkout),
```

Comment:
Now, that is elegant! I wonder however if `select`'s scope should just be about selecting items from the State of the context, and not provide additional props which can be inferred from it. Maybe this kind of computations should stay within the component. WDYT?

Reply:
I don't think it makes any difference in this case, but I don't think we should limit it. 

There could be a use case if in various class functions we would use daysOfStay props, it is better to have it generated and passed in like this I guess.

Reply:
True, and even that logic could be integrated into the context.